torchrl/collectors/distributed/generic.py

```diff
@@ -0,0 +1,831 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""Generic distributed data-collector using torch.distributed backend."""
+
+import os
+import socket
+import warnings
+from copy import copy, deepcopy
+from datetime import timedelta
+from typing import OrderedDict
+
+import torch.cuda
+from tensordict import TensorDict
+from torch import multiprocessing as mp, nn
+
+from torchrl.collectors import MultiaSyncDataCollector
+from torchrl.collectors.collectors import (
```

Comment:
Could consider making `_DataCollector` public.

Reply:
Yep
`DataCollectorBase` would be nice!